docs/code-quality/advanced.md

```diff
@@ -0,0 +1,129 @@
+# Advanced
+
+## Table of contents
+- [Advanced](#advanced)
+	- [Table of contents](#table-of-contents)
+	- [Large Update Rules](#large-update-rules)
+	- [Add Comments](#add-comments)
+	- [Write Tests](#write-tests)
+	- [Force Push](#force-push)
+	- [Don't Overengineer](#dont-overengineer)
+	- [Use Existing Code](#use-existing-code)
+	- [Understand Impact](#understand-impact)
+	- [Suggesting A Library](#suggesting-a-library)
+	- [Knowing What You Don't Know](#knowing-what-you-dont-know)
+	- [Aware of Impact](#aware-of-impact)
+
+## Large Update Rules
+
+Any issue that requires more than 10 commits or 50 lines of code can be considered a large update (unless it is pure refactor)
+
+1. If you're doing a big feature, first create a branch and do a rough version of the feature that works, without worrying much about commit readability and architecture
+2. Push that branch to Github as a draft PR
+3. Review your commit at PR UI from the eye of reviewer 
+4. Give yourself stuff to fix, like change var name, separate variables, split function into multiple, do better architecture, etc..
+5. Then create a todo list, and how could the architecture have been done cleanly
+6. Create a new branch
+7. Start moving code to new branch and rewrite everything but in a clean, final way
+8. If this feature will require multiple PRs, then create a main PR, then write code in parts in the PRs.
+	- Example: https://prnt.sc/2011xo4
+	- Video explanation: https://www.loom.com/share/703185b81607482ea304838064d34578
+9. If a code/logic is hard to understand first create a PR that refactors that part of the code, once its approved create the new feature
+10. If you have to merge development branch please make sure to follow following. It will help to track commits properly
+    - squash merge them
+	- merge it in the main branch only
+	- if you need development in sub branch, merge it in the main branch first and then create a sub branch from it
+	- Bad example: https://prnt.sc/9gbezql7iQ0P
+
+## Add Comments
+
+We try to avoid comments as much as possible but under certain circumstances they need to be added.
+Temporary, Optional & Confusing codes need explanation.
+Leave a comment like this: https://prnt.sc/20125e9
+
+## Write Tests
+
+Complex logic should always be tested 
+Examples: https://prnt.sc/20127k6 https://prnt.sc/20127s7
+
+## Force Push
+
+- Do not force push on branches that have received feedback
+- Instead fork your previous branch, create a new feedback branch and do the updates there and merge into your original branch. https://www.loom.com/share/9bf9e79e21bc442498182571c6d62517
```

Comment:
Instead, create a feedback branch from your main branch, and commit your updates there. Then merge the feedback branch into your main branch.